### DISCUSSION TITLE
Programmatically Create Note In Current Workspace

### DISCUSSION THREAD

**Discussion by walkerad03** [TriliumNext/Trilium#3596] [Q&A]
Hi all,
I am currently trying to create a single button I can use to create templated notes for my classes. I use workspaces to keep all of my work for each of my classes separate; 1 workspace for each class. However, I would like to use a single button to create a new note for whichever workspace I currently have open.

Here is the code I have so far:
```
api.addButtonToToolbar({
    title: "New Lecture Note",
    icon: 'edit',
    shortcut: 'alt+n',
    action: async() => {
        const rootNote = await api.getActiveContextNote().noteId;
        const newNote = await api.runOnBackend((rootNote) => {
            const template = api.getNoteWithLabel('lectureNoteTemplate');
            
            const newNote = api.createTextNote(rootNote, 'Lecture ');
            newNote.note.setRelation('template', template.noteId);
            
            return newNote.note;
        }, [rootNote]);
    
        
        await api.waitUntilSynced();
        await api.activateNewNote(newNote.noteId);
       
        const dateHuman = api.dayjs().format('dddd, MMMM D, YYYY')
        const dateISO = api.dayjs().format('YYYY-MM-DD');
        await api.addTextToActiveContextEditor("Date: " + dateISO + " (" + dateHuman + ")");
    }
});
```
As you can see, I am finding the rootNote for the lectures by calling the frontend `api.getActiveContextNote();`. I would rather automatically select the root by using a label such as `#lectureRoot` through the backend. When I have tried this by calling
```
const rootNote = await api.getNoteWithLabel('lectureRoot');
```
I have found that Trilium does correctly select a note with that label, however the note is located in a different, unhoisted workspace. Is there any way to search for notes that are only present in the currently hoisted workspace?

Thanks in advance.

**Comment by walkerad03** [comment]
Update: I have come up with a bit of a sketchy solution for the mean time by getting the parent of each of the notes tagged with `#lectureNote` and then comparing it to the currently active note which will be the root of the workspace right after activating said workspace. This fails if I have any other note selected when I press the button, but it is a step closer to what I am looking for. Code is here:
```
api.addButtonToToolbar({
    title: "New Lecture Note",
    icon: 'edit',
    shortcut: 'alt+n',
    action: async() => {
        const rootNote = await api.getActiveContextNote().noteId;
        const newNote = await api.runOnBackend((rootNote) => {
            const lectureRoots = api.getNotesWithLabel('lectureRoot');
            const template = api.getNoteWithLabel('lectureNoteTemplate');
            for (var i = 0; i < lectureRoots.length; i++) {
                const parentNote = lectureRoots[i].getParentNotes();
                if (parentNote[0].noteId == rootNote) {
                    const newNote = api.createTextNote(lectureRoots[i].noteId, 'New Lecture');
                    newNote.note.setRelation('template', template.noteId);
                    return newNote.note;
                    break;
                }
            }
        }, [rootNote]);
        
        await api.waitUntilSynced();
        await api.activateNewNote(newNote.noteId);
       
        const dateHuman = api.dayjs().format('dddd, MMMM D, YYYY')
        const dateISO = api.dayjs().format('YYYY-MM-DD');
        await api.addTextToActiveContextEditor("Date: " + dateISO + " (" + dateHuman + ")");
    }
});
```
I am hoping to find a legitimate solution that could let me just select the workspace root or something similar.

**Comment by zadam** [comment]
Hi, try `searchForNotes` which is more flexible:

```javascript
api.searchForNotes('#lectureRoot', {
    ignoreHoistedNote: false
});
```

This should search within the currently hoisted tree (= workspace).
